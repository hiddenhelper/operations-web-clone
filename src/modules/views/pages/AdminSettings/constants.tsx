import React from 'react';
import { ProcoreIcon } from '../../../../constants';

export const titleProcoreCard = 'Procore Integration';
export const bodyProcoreCardFirstLine = 'FCA Procore integration allows for accurate daily';
export const bodyProcoreCardSecondLine = 'census data via automated mapping of clients.';
export const logoProcore = <ProcoreIcon fill="#1E86E8" height="38px" width="38px" />;
export const linkDescriptionProcore = 'Map Clients with Procore';
