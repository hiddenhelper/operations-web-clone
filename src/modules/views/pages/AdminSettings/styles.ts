import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../constants';
import { toREM } from '../../../../utils/generalUtils';

export const useStyles = makeStyles(theme =>
  createStyles({
    cardsWrapper: { marginTop: 40 },
    adminName: {
      fontSize: `${toREM(STYLE.FONT.SIZE.HEADING_BIG)} !important`,
      marginBottom: '30px',
    },
    adminTitleWrapper: {
      width: '100%',
      display: 'flex',
      justifyContent: 'space-between',
      alignItems: 'center',
    },
    adminText: {
      color: STYLE.COLOR.SECONDARY_DARKER,
      fontSize: `${toREM(31)}`,
      letterSpacing: 0,
    },
    adminAccent: {
      fontWeight: 600,
    },
    adminTitleMarginRight: {
      marginRight: `${toREM(12)}`,
    },
    chipSize: {
      width: '70px',
      height: '20px !important',
      fontSize: '12px !important',
    },
    tableContainer: {
      '& .MuiTableCell-root': {
        height: 60,
      },
    },
    alignLeftStatus: {
      paddingRight: '42px !important',
    },
  })
);
