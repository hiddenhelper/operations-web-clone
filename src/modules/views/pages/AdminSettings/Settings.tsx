import React, { memo } from 'react';
import Container from 'modules/views/shared/Container';
import PageTitle from 'modules/views/shared/PageTitle';
import { CardAdmin } from './components/CardAdmin';
import { useStyles } from './styles';
import { titleProcoreCard, bodyProcoreCardFirstLine, bodyProcoreCardSecondLine, logoProcore, linkDescriptionProcore } from './constants';

const AdminSettings = () => {
  const classes = useStyles();
  return (
    <Container>
      <PageTitle title="System Administration" />
      <div className={classes.cardsWrapper}>
        <CardAdmin
          title={titleProcoreCard}
          bodyFirstLine={bodyProcoreCardFirstLine}
          bodySecondLine={bodyProcoreCardSecondLine}
          logo={logoProcore}
          linkDescription={linkDescriptionProcore}
        />
      </div>
    </Container>
  );
};

export default memo(AdminSettings);
