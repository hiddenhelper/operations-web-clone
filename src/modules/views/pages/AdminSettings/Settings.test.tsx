import React from 'react';
import { render } from '@testing-library/react';
import AdminSettings from './Settings';
import { MemoryRouter } from 'react-router';

describe('Sidebar', () => {
  it('should render admin tab', () => {
    const { container } = render(
      <MemoryRouter>
        <AdminSettings />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
