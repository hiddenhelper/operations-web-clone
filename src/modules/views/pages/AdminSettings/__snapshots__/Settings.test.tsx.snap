// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render admin tab 1`] = `
<div>
  <div
    style="width: 100%; background-color: rgb(245, 245, 245); padding: 2.5rem;"
  >
    <div
      class="MuiBox-root MuiBox-root-10"
      style="height: 50px; margin-bottom: 13px; width: 100%;"
    >
      <div
        class="MuiBox-root MuiBox-root-11"
        style="display: flex; align-items: center; justify-content: space-between;"
      >
        <div>
          <div
            class="MuiBox-root MuiBox-root-12"
            style="display: flex; align-items: center; margin-top: 5px;"
          >
            <h2
              class="MuiTypography-root MuiTypography-h1"
              style="color: rgb(68, 68, 68); font-size: 2.25rem; font-weight: 600; letter-spacing: 0; margin-right: 0.5rem;"
            >
              System Administration
            </h2>
          </div>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-cardsWrapper-1"
    >
      <div
        class="makeStyles-cardContainer-13"
        data-testid="procoreCard"
      >
        <div
          class="makeStyles-titleContainer-14"
        >
          <div
            class="makeStyles-logoContainer-15"
          >
            <div
              class="makeStyles-grayCircle-16"
            >
              <svg
                fill="#1E86E8"
                height="38px"
                width="38px"
              >
                procore.svg
              </svg>
            </div>
          </div>
          <div>
            <p
              class="MuiTypography-root makeStyles-title-17 MuiTypography-body1"
            >
              Procore Integration
            </p>
          </div>
        </div>
        <div
          class="makeStyles-bodyContainer-18"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            FCA Procore integration allows for accurate daily
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            census data via automated mapping of clients.
          </p>
        </div>
        <div
          class="makeStyles-linkContanier-19"
        >
          <a
            color="inherit"
            href="/procore-clients"
          >
            Map Clients with Procore
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
